core/src/lib.rs

```diff
@@ -2,3 +2,30 @@ pub mod config;
 pub mod public_key;
 
 pub use config::Config;
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn hc_public_key_url() {
+        let email: String = "pj@aa.pl".to_string();
+        let password: String =  "password".to_string();
+        let seed: Option<[u8; 32]> = Some([55; 32]);
+        let expected_url: String = "https://hcscimeesmngnuygkhtit5auwbfiuivxjmff7o54speb6zg84yebxuv7bf7z58z.holohost.net/".to_string();
```

Comment:
```suggestion
        let expected_url = "https://hcscimeesmngnuygkhtit5auwbfiuivxjmff7o54speb6zg84yebxuv7bf7z58z.holohost.net/";
```

Reply:
Also, I'd prefer if you'd convert this to `Url` type in-place and compare with `Url` below without transforming it into a string.

Reply:
One way or another types have to be converted to the same ones.